server/handlers.go

```diff
@@ -309,3 +311,47 @@ func signMessageHandler(client eth.LivepeerEthClient) http.Handler {
 		w.Write(signed)
 	})
 }
+
+func voteHandler(client eth.LivepeerEthClient) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if client == nil {
+			respondWith500(w, "missing ETH client")
+			return
+		}
+
+		poll := r.FormValue("poll")
+		if poll == "" {
+			respondWith500(w, "missing poll contract address")
+			return
+		}
+		if !ethcommon.IsHexAddress(poll) {
+			respondWith500(w, "invalid poll contract address")
+			return
+		}
+
+		choiceStr := r.FormValue("choiceID")
+		if choiceStr == "" {
+			respondWith500(w, "missing choiceID")
+			return
+		}
+
+		choiceID, _ := new(big.Int).SetString(choiceStr, 10)
```

Comment:
We should check the boolean here because if `choiceStr` is not a valid integer then `choiceID` will be nil and L340 will trigger a nil pointer error.

Reply:
fixed in 2857efe